api/resolvers/item.js

```diff
@@ -1363,6 +1385,38 @@ const enqueueDeletionJob = async (item, models) => {
   }
 }
 
+const deleteReminderAndJob = async ({ me, item, models }) => {
+  if (me?.id && me.id !== ANON_USER_ID) {
+    await models.$queryRawUnsafe(`DELETE FROM pgboss.job WHERE name = 'reminder' AND data->>'itemId' = '${item.id}' AND data->>'userId' = '${me.id}' AND state <> 'completed';`)
+    await models.reminder.deleteMany({
+      where: {
+        itemId: Number(item.id),
+        userId: Number(me.id),
+        remindAt: {
+          gt: new Date()
+        }
+      }
+    })
+  }
+}
+
+const createReminderAndJob = async ({ me, item, models }) => {
+  // disallow anon to use reminder
+  if (!me || me.id === ANON_USER_ID) {
+    return
+  }
+  const reminderCommand = getReminderCommand(item.text)
+  if (reminderCommand) {
+    await models.$queryRawUnsafe(`
+      INSERT INTO pgboss.job (name, data, startafter)
+      VALUES ('reminder', jsonb_build_object('itemId', ${item.id}, 'userId', ${me.id}), now() + interval '${reminderCommand.number} ${reminderCommand.unit}s');`)
```

Comment:
We'll need to set an `expiresin` here. By default this job will be ignored after 15 minutes: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#job-table

I noticed this when I was queuing `territoryBilling` jobs that `startafter` 1 month or 1 year.

Reply:
I looked at the documentation now, especially since you also mentioned this to me. But afaict, `expireIn` is only relevant for active jobs so this shouldn't affect billing jobs that _start_ (and _then_ are active) in a month or year?

That would also match my expectations, why would pgboss drop jobs that haven't even started yet?

[See documentation for job states](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#job-states):

> All jobs start out in the created state and become active when picked up for work. If job processing completes successfully, jobs will go to completed. If a job fails, it will typically enter the failed state. However, if a job has retry options configured, it will enter the retry state on failure instead and have a chance to re-enter active state. It's also possible for active jobs to become expired, which happens when job processing takes too long. Jobs can also enter cancelled state via [cancel(id)](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#cancelid) or [cancel([ids])](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#cancelids).

> **Expiration options**
> - expireInSeconds, number
How many seconds a job may be in active state before it is failed because of expiration. Must be >=1
>
> [...]
> 
> Default: 15 minutes

So what you're describing sounds like a bug

Reply:
That's a good point. I think I borrowed this syntax from the delete bot job, which I don't think had it either? I could be wrong. But I agree, it does seem like a bug if you need to set expiration for a job that starts more than 15 minutes in the future. I don't know if I explicitly tested a reminder that far in the future so I can't say if it was an issue here.